resources/virt.go

```diff
@@ -108,15 +143,71 @@ func (obj *VirtRes) Init() error {
 
 	obj.absent = (obj.Transient && obj.State == "shutoff") // machine shouldn't exist
 
+	obj.conn, err = obj.connect() // gets closed in Close method of Res API
+	if err != nil {
+		return errwrap.Wrapf(err, "%s[%s]: Connection to libvirt failed in init", obj.Kind(), obj.GetName())
+	}
+
+	// check for hard to change properties
+	dom, err := obj.conn.LookupDomainByName(obj.GetName())
+	if err == nil {
+		defer dom.Free()
+	} else if !isNotFound(err) {
+		return errwrap.Wrapf(err, "%s[%s]: Could not lookup on init", obj.Kind(), obj.GetName())
+	}
+
+	if err == nil {
```

Comment:
@ffrank That block was stuff we only do when we have a working machine. If we init and the machine doesn't exist yet, then we don't need to query it to get initial state.